### PR TITLE
Home work_Template pattern

### PR DESCRIPTION
**Pull request by DadizhaO**

### CODE REVIEW

File: src/test/java/HM19/OfficeDaoImplGenericTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class OfficeDaoImplGenericTest {
+
+    private OfficeDaoImplGeneric officeDaoImplgeneric = new OfficeDaoImplGeneric();
+
+    private Office office1 = new Office(new BigDecimal(120), "city", "region",
+            null, null, new BigDecimal(1223));
+    private Office office2 = new Office(new BigDecimal(120), "city2", "region2",
+            null, new BigDecimal(10101), new BigDecimal(20202));
+
```
Comment (u2107680007911543774e7r) [inline src/test/java/HM19/OfficeDaoImplGenericTest.java:25]: These fields are meant to be static I think

**Review by u2107680007911543774e7r — APPROVED** [review]
Good.

File: src/main/java/HM19/OfficeDaoImplGeneric.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class OfficeDaoImplGeneric implements OfficeDao {
```
Comment (SerhiiPasko) [inline src/main/java/HM19/OfficeDaoImplGeneric.java:15]: Bad name

File: src/main/java/HM19/TemplateAbs.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package HM19;
+
+import Lesson15.ConnectToDB;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public abstract class TemplateAbs<T> {
```
Comment (SerhiiPasko) [inline src/main/java/HM19/TemplateAbs.java:9]: bad name

**Review by SerhiiPasko — APPROVED** [review]
Good. Read comments

**Pull request merged**